File: generator/src/main/java/ch/bailu/gtk/model/JavaType.java
package ch.bailu.gtk.model;

import ch.bailu.gtk.table.PrimitivesTable;
import ch.bailu.gtk.tag.ParameterTag;

public class JavaType {

    private final String type;


    public JavaType(ParameterTag parameter) {
	    this(parameter.getType());
    }
    
    public JavaType(String type) {
        this.type = setType(PrimitivesTable.instance().convert(type));
    }


    private static String setType(String type) {
        if (type == null) {
            type = "";
        }
        return type;
    }

    public boolean isValid() {
        return ! "".equals(type);
    }

    public String getType() {
        return type;
    }

    public boolean isVoid() {
        return "void".equals(type);
    }
}

File: generator/src/main/kotlin/ch/bailu/gtk/model/type/JavaType.kt
package ch.bailu.gtk.model.type

import ch.bailu.gtk.table.PrimitivesTable

/**
 * A Primitive java type.
 * Is valid if name is configured in [PrimitivesTable]
 */
class JavaType(typeName: String) {
    private var type = PrimitivesTable.convert(typeName)

    val valid: Boolean
        get () = type.isNotEmpty()

    fun getApiTypeName(): String {
        return type
    }

    fun isVoid(): Boolean {
        return "void" == type
    }
}
